### DISCUSSION TITLE
SEC_ERROR_INADEQUATE_KEY_USAGE  On HTTPS all of the sudden SSL activated for about a year...

### DISCUSSION THREAD

**Discussion by gcyrillus** [leokhoa/laragon#423] [Q&A]
Hello, i had SSL activated for months , untill suddenly Firefox tells me it is dangerous, so i clicked okay to keep going on and that worked for a few days.

Now i get this error message anytime i try to use https :  SEC_ERROR_INADEQUATE_KEY_USAGE  for FF and   NET::ERR_CERT_AUTHORITY_INVALID for chrome and says that my OS (win10 64bits) doesn't recognise it as safe. I still can keep going on with chrome.

I tried to disabled and able SSL again , but it had no effects and i have no idea how i can get laragon to renew or create a new and valid certificat.

Thanks for reading and sharing your possible fix or explanations.

take care all
GC

**Comment by Pen-y-Fan** [comment]
My colleague had a similar problem, adding laragon.crt to the trusted store worked for him. I have a blog with more details:

- [How to set up Laragon on a new Windows computer (part 4) - Enable security](https://pen-y-fan.github.io/2022/11/10/how-to-set-up-laragon-on-a-new-windows-computer-part-4/)

**Reply by gcyrillus** [reply]
it worked for about three weeks . Then again, 
FF:  SEC_ERROR_INADEQUATE_KEY_USAGE
Chrome :   ERR_SSL_KEY_USAGE_INCOMPATIBLE

**Comment by gcyrillus** [comment]
@Pen-y-Fan 
Perfect, i should have digged a little more into laragon's menus. _(Add laragon.crt to Trusted Store )_
Thank's a lot, cheers

**Comment by gcyrillus** [comment]
Trouble is all of the sudden back again.  Is there any other stable way to fix this ?

**Comment by rubendn** [comment]
I was able to fix it on my machine running Windows 11 and Laragon 6.0 by doing the following:

1) Modify the following file:  \laragon\usr\tpl\openssl.conf.tpl
    Change the line that reads:

        keyUsage = keyEncipherment, dataEncipherment

    TO:

        keyUsage = digitalSignature, keyEncipherment

2) Stop and Restart Laragon
3) Menu->Apache->SSL-> Add laragon.crt to Trust Store
4) Stop and Restart Laragon

I believe what is happening is that when Laragon starts it is writing laragon\etc\ssl\auto.openssl.conf based on the template file in Step 1.  If you only modify  laragon\etc\ssl\auto.openssl.conf, it will get overwritten the next time Laragon starts.

After these steps, I'm able to use SSL in both Chrome and Edge without any warnings.

**Reply by gcyrillus** [reply]
Hello, thanks for the tip,  i did not have that line to edit, so i added it.
It seems to work for Chrome and edge indeed, but for Firefox , no change, i still get that warning ! even that i added manually the certificates via the firefox config (funny thing is that when i do so, then no more warnings from other browsers) .
If i try to add it again, because i closed FF and restarted the computer next day, it says it is already there and valid! but i have to go throught the warning and click on to keep going.
I run W10  [Next is out of subject - don't mind it this too much or not at all]  W11 ?  @MS why should I update and loose a few things, mess up my config  and probably slow down the computer?

**Reply by rubendn** [reply]
Hi, I still get the warning on FF but it works.  I do not get a warning on Chrome or Edge.

**Reply by Pen-y-Fan** [reply]
On firefox can "accept the risk", see my blog [How to set up Laragon on a new Windows computer (part 4) - Enable security](https://pen-y-fan.github.io/2022/11/10/how-to-set-up-laragon-on-a-new-windows-computer-part-4/)

**Comment by gcyrillus** [comment]
@rubendn well , it finally broke again! It seems that it happens if i add a PHP version to laragon then, it breaks as soon as i reboot the computer.

Actually this helped me too  : https://getyourbitstogether.com/fixing-sec_error_inadequate_key_usage/  (2files 2 delete, how easy!) Of course, it is brutal and doesn't really cure anything.

@Pen-y-Fan  "accept the risk" is not always an option that is provided by the browser , that is my case most of the time. even if you gave me twice the link ;) 

@leokhoa I have no idea how i could apply this : https://letsencrypt.org/docs/certificates-for-localhost/ so our laragon local-server do not make browser choke anymore ;) Would it be efficient and good enough to replace the default certificate built-in laragon?


(w10 laragon full 6.0 220616  php : 7.2.19/7.3.2/8.1.10/8.1.5/8.2.5/8.2.6  apache 2.4.54  )
